lock-api/src/main/java/com/palantir/lock/client/UnreliableTimeLockService.java

```diff
@@ -0,0 +1,147 @@
+    private final BuggifyFactory buggify;
+
+    public static UnreliableTimeLockService create(TimelockService timelockService) {
+        return new UnreliableTimeLockService(timelockService, DefaultBuggifyFactory.INSTANCE);
+    }
+
+    @VisibleForTesting
+    UnreliableTimeLockService(TimelockService delegate, BuggifyFactory buggifyFactory) {
+        this.delegate = delegate;
+        this.buggify = buggifyFactory;
+    }
+
+    @Override
+    public boolean isInitialized() {
+        return delegate.isInitialized();
+    }
+
+    @Override
+    public long getFreshTimestamp() {
+        return delegate.getFreshTimestamp();
+    }
+
+    @Override
+    public long getCommitTimestamp(long startTs, LockToken commitLocksToken) {
+        return delegate.getCommitTimestamp(startTs, commitLocksToken);
+    }
+
+    @Override
+    public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
+        return delegate.getFreshTimestamps(numTimestampsRequested);
+    }
+
+    @Override
+    public LockImmutableTimestampResponse lockImmutableTimestamp() {
+        return delegate.lockImmutableTimestamp();
+    }
+
+    @Override
+    public List<StartIdentifiedAtlasDbTransactionResponse> startIdentifiedAtlasDbTransactionBatch(int count) {
+        return delegate.startIdentifiedAtlasDbTransactionBatch(count);
+    }
+
+    @Override
+    public long getImmutableTimestamp() {
+        return delegate.getImmutableTimestamp();
+    }
+
+    @Override
+    public LockResponse lock(LockRequest request) {
+        LockResponse response = delegate.lock(request);
+        buggify.maybe(0.25).run(() -> {
+            log.info("BUGGIFY: Unlocking lock token {} after acquiring", SafeArg.of("token", response.getToken()));
+            delegate.unlock(Set.of(response.getToken()));
+        });
+        return response;
+    }
+
+    @Override
+    public LockResponse lock(LockRequest lockRequest, ClientLockingOptions options) {
+        LockResponse response = delegate.lock(lockRequest, options);
+        buggify.maybe(0.25).run(() -> {
+            log.info("BUGGIFY: Unlocking lock token {} after acquiring", SafeArg.of("token", response.getToken()));
+            delegate.unlock(Set.of(response.getToken()));
+        });
+        return response;
+    }
+
+    @Override
+    public WaitForLocksResponse waitForLocks(WaitForLocksRequest request) {
+        return delegate.waitForLocks(request);
+    }
+
+    @Override
+    public Set<LockToken> refreshLockLeases(Set<LockToken> tokens) {
+        Set<LockToken> tokensToRefresh = tokens.stream()
+                .filter(_token -> !buggify.maybe(0.10).asBoolean())
+                .collect(Collectors.toSet());
+        Set<LockToken> tokensToUnlock = Sets.difference(tokens, tokensToRefresh);
+        if (!tokensToUnlock.isEmpty()) {
+            log.info("BUGGIFY: Unlocking tokens on refresh: {}", SafeArg.of("tokens", tokensToUnlock));
+            unlock(tokensToUnlock);
```

Comment:
nit: `delegate.unlock(tokensToUnlock);` - while I know they're equivalent here I don't know if we necessarily want to include nested buggification :)

(or if this is intended then the lock ones should probably also call unlock directly?)

Reply:
I was biasing towards "we should call the unlock method in our class so we don't accidentally bypass behavior of this class", but in this context, it probably makes more sense for us to call the delegate, as we probably do want to bypass behavior of this class!

Reply:
Yep! In a majority of cases you do want to call the local as opposed to the delegate when invoking other methods, though this one is a bit special :)